script.js

```diff
@@ -0,0 +1,158 @@
+      document.getElementById("password").style.borderColor = "red";
+      alert(
+        "password should contain atlest 8characters including one uppercase,number and special character"
+      );
+      document.getElementById("password").style.borderColor = "white";
+      return false;
+    }
+
+    if (!phoneNumberValid(phoneNumber) || !phonenumNotExist(phoneNumber)) {
+      document.getElementById("phonenumber").style.borderColor = "red";
+      alert("phone number should contain 10 numbers");
+      document.getElementById("phonenumber").style.borderColor = "white";
+      return false;
+    }
+
+    addDetails(userName, email, phoneNumber, encryptedPassword);
+    document.getElementById("signup").reset();
+    return true;
+  }
+
+  function validateDetails(event) {
+    event.preventDefault();
+    const loginEmail = document.getElementById("loginEmail").value;
+    const loginPassword = document.getElementById("loginPassword").value;
+    const userExist = userDetails.filter((obj) => obj.userEmail == loginEmail);
+    const decryptPassword = crypt.decrypt(userExist[0].userPassword);
+
+    if (decryptPassword === loginPassword) {
+      const user = userExist[0].userName;
+      setCookie("userName", user, 1);
+      document.getElementById("login").reset();
+      window.location.href = "/login/dashboard.html";
+    } else {
+      alert("password is incorrect");
+    }
+  }
+
+  function userNameValid(userName) {
+    const userNamePattern = /^[a-zA-Z ]{3,30}$/;
+    return userNamePattern.test(userName);
+  }
+
+  function emailValid(email) {
+    const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+    return emailPattern.test(email);
+  }
+
+  function passwordValid(password) {
+    return (
+      /[!@#$%^&*()_+{}\[\]:;<>,.?~\-]/.test(password) &&
+      /[A-Z]/.test(password) &&
+      /\d/.test(password)
+    );
+  }
+
+  function phoneNumberValid(phoneNumber) {
+    const numberPattern = /^\d{10}$/;
+    return numberPattern.test(phoneNumber);
+  }
+
+  function emailNotExist(email) {
+    const userExist = userDetails.findIndex((obj) => obj.userEmail === email);
+    if (userExist !== -1) {
+      alert("email already exist");
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function phonenumNotExist(phoneNumber) {
+    const userExist = userDetails.findIndex(
+      (obj) => obj.userNumber === phoneNumber
+    );
+    if (userExist !== -1) {
+      alert("phonenumber already exist");
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function setCookie(name, userNameid, dayslive) {
+    const todate = new Date();
+    todate.setTime(todate.getTime() + dayslive * 24 * 60 * 60 * 1000);
+    let expiryDate = "expires=" + todate.toUTCString();
+    document.cookie = `${name}=${userNameid};${expiryDate}; path/`;
+  }
+
+  function addDetails(userName, email, phoneNumber, password) {
+    let id = userDetails.length + 1;
+    newUser = {
+      id: id,
```

Comment:
Instead of   id: id,
```suggestion
      id,
```